formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/CommentLexers.kt

```diff
@@ -0,0 +1,197 @@
+                    }
+                }
+            }
+            break
+        }
+        currentPosition = current
+        return current
+    }
+}
+
+internal class ReaderJsonLexerWithComments(reader: InternalJsonReader, buffer: CharArray): ReaderJsonLexer(reader, buffer) {
+    override fun consumeNextToken(expected: Char) {
+        ensureHaveChars()
+        val source = source
+        val current = skipWhitespaces()
+        // skipWhitespaces() calls prefetch() on every iteration, so we can be sure that there's at least THRESHOLD-1 chars in buf when it returns.
+        if (current >= source.length || current == -1) {
+            currentPosition = -1 // for correct EOF reporting
+            unexpectedToken(expected) // EOF
+        }
+        val c = source[current]
+        currentPosition = current + 1
+        if (c == expected) return
+        else unexpectedToken(expected)
+    }
+
+    override fun canConsumeValue(): Boolean {
+        ensureHaveChars()
+        val current = skipWhitespaces()
+        // skipWhitespaces() calls prefetch() on every iteration, so we can be sure that there's at least THRESHOLD-1 chars in buf when it returns.
+        if (current >= source.length || current == -1) return false
+        return isValidValueStart(source[current])
+    }
+
+    override fun consumeNextToken(): Byte {
+        ensureHaveChars()
+        val source = source
+        val cpos = skipWhitespaces()
+        if (cpos >= source.length || cpos == -1) return TC_EOF
+        currentPosition = cpos + 1
+        return charToTokenClass(source[cpos])
+    }
+
+    override fun peekNextToken(): Byte {
+        ensureHaveChars()
+        val source = source
+        val cpos = skipWhitespaces()
+        // skipWhitespaces() calls prefetch() on every iteration, so we can be sure that there's at least THRESHOLD-1 chars in buf when it returns.
+        if (cpos >= source.length || cpos == -1) return TC_EOF
+        currentPosition = cpos // only difference with consumeNextToken(), actually
+        return charToTokenClass(source[cpos])
+    }
+
+    private fun handleComment(position: Int): Pair<Int, Boolean> {
+        var current = position
+        var startIndex = current + 2
+        when (source[current + 1]) {
+            '/' -> {
+                while(current != -1) {
+                    current = source.indexOf('\n', startIndex)
+                    if (current == -1) {
+                        current = prefetchOrEof(source.length)
+                        startIndex = current
+                    } else {
+                        return current + 1 to true
+                    }
+                }
+                // reached end of stream.
+                return -1 to true
+            }
+
+            '*' -> {
+                while(current != -1) {
+                    current = source.indexOf("*/", startIndex)
+                    if (current == -1) {
+                        current = prefetchOrEof(source.length)
```

Comment:
What if the last char in the current source was `'*'`, and after `prefetchOrEof` there is a `'/'` at the beginning? 
In this case, we skip the closing of the comment.

Reply:
Good catch. Added a test for that